### PR TITLE
feat: add buf linter and formater

### PR DESCRIPTION
**Pull request by Cluas**

### CODE REVIEW

File: lua/null-ls/builtins/diagnostics/buf.lua [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+
+local DIAGNOSTICS_ON_SAVE = methods.internal.DIAGNOSTICS_ON_SAVE
+
+return h.make_builtin({
+    name = "buf_lint",
+    meta = {
+        url = "https://github.com/bufbuild/buf",
+        description = "A new way of working with Protocol Buffers.",
+    },
+    method = DIAGNOSTICS_ON_SAVE,
```
Comment (jose-elias-alvarez) [inline lua/null-ls/builtins/diagnostics/buf.lua:12]: I always want to check this, but: Is there a specific reason for this linter to run on save only? It shouldn't be necessary if it accepts `stdin`.

**Review by jose-elias-alvarez — COMMENTED** [review]
Looks good! Just want to clarify if it's possible to run this on save, but if not let me know.

File: lua/null-ls/builtins/diagnostics/buf.lua (hunk 1 above)
Comment (Cluas) [reply lua/null-ls/builtins/diagnostics/buf.lua]: Your are right, indeed there is no effect, directly using DIAGNOSTICS

**Comment by jose-elias-alvarez** [comment]
Excellent, thank you for the contribution! For future reference, the documentation is generated automatically on merge, so no need to do it manually.

**Pull request merged**